prboom2/src/SDL/i_sound.c

```diff
@@ -163,21 +164,70 @@ static void stopchan(int i)
 //
 static int addsfx(int sfxid, int channel, const unsigned char *data, size_t len)
 {
+  channel_info_t *ci = channelinfo + channel;
+
   stopchan(channel);
 
-  channelinfo[channel].data = data;
-  /* Set pointer to end of raw data. */
-  channelinfo[channel].enddata = channelinfo[channel].data + len - 1;
-  channelinfo[channel].samplerate = (channelinfo[channel].data[3] << 8) + channelinfo[channel].data[2];
-  channelinfo[channel].data += 8; /* Skip header */
+  if (len > 44 && memcmp(data, "RIFF", 4) == 0 && memcmp(data + 8, "WAVEfmt ", 8) == 0)
+  {
+    SDL_RWops *RWops;
+    SDL_AudioSpec wav_spec;
+    Uint8 *wav_buffer = NULL;
+    int bits, samplelen;
+
+    RWops = SDL_RWFromMem(data, len);
+
+    Z_ChangeTag(data, PU_CACHE);
```

Comment:
What's this line about? All sound lumps in dsda-doom are locked at launch.

Reply:
`SDL_LoadWAV_RW` loads the data into memory and decodes it if necessary, so the original data buffer is no longer used. Woof can use `Z_ChangeTag(data, PU_CACHE)` here, not sure about PrBoom+.
The reason for using SDL is because the WAV format is complex and there are many problematic WAV files (pistol sound in UACrebellion.wad for example).

Reply:
Fixed in https://github.com/kraflab/dsda-doom/pull/110/commits/3fc17ea07dddb481efa3648375848a8585200464